compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -610,11 +623,22 @@ string t_py_generator::render_const_value(t_type* type, t_const_value* value) {
   return out.str();
 }
 
+/** 
+ * Generates the "forward declarations" for python structs.
+ * These are actually full class definitions so that calls to generate_struct
+ * can add the thrift_spec field.  This is needed so that all thrift_spec 
+ * definitions are grouped at the end of the file to enable co-recursive structs.
+ */
+void t_py_generator::generate_forward_declaration(t_struct* tstruct) {
```

Comment:
This is an interface method that's called by the thrift compiler.  One attribute of this change is to move most of the code that generates python code into ``generate_forward_declaration`` and then output the ``thrift_spec`` via the ``generate_struct`` method.